enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+For an explicit signal about the necessity to set `--cloud-provider=external` flag to the kubelet, we will use the `CloudControllerManager` field of the `ExternalPlatformSpec`, which is  
+described in the [API Extensions](#api-extensions) section down below.
+
+#### Kube Controller Manager Operator
+
+Same as the [MCO](#machine-config-operator) in regard to kubelet, Kube Controller Manager Operator manages KCM (kube-controller-manager) deployments.
+
+Historically Kube Controller Manager was home for cloud-specific control loops. This logic is also engaging by setting up proper
+flags on KCM executable like,
+```shell
+...
+--cloud-provider=azure
+--cloud-config=/etc/kubernetes/cloud.conf
+...
+```
+
+For engaging an external Cloud Controller Manager no `cloud-provider` flag should be set for the KCM executable.
+
+In the context of this EP, no particular changes will be needed in the operator itself.
+
+#### Cluster Cloud Controller Manager Operator
+
+Responsible for deploying platform-specific Cloud Controller Manager as well as for handling a number of OpenShift's
+specific peculiarities (such as populating proxy settings for CCMs, sync credentials and so on).
+
+During the first phases of the "External" platform type enablement, this operator should be just disabled.
+In the future, when the delivery mechanism for CCMs will be defined, the operator might be engaged for deploying a user-provided cloud controller manager,
+however this is a subject for the upcoming design work. 
+
+#### Library Go
+
+Significant part of the code around PlatformType handling lives in the ["openshift/library-go"](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go).
+
+For the moment, this code is responsible for the decision around kubelet and kcm flags. Precisely, [IsCloudProviderExternal](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L154) function
+is used for decisions around kubelet and KCM flags (within MCO and KCMO respectively). Also, this code is used for the decision-making about CCM operator engagement.
+
+#### Machine Api Operator
+
+Machine Api Operator is responsible for deploying and maintaining the set of machine-api related controllers, such as
+
+- machineset controller
+- nodelink controller
+- machine health check controller
+- machine controller
+
+From the list above, the only "machine" controller is cloud-provider dependent, however, for now, Machine Api Operator
+won't deploy anything if it encounters "None" or an unrecognized platform type. 
+
+In the future, "External" platform type would serve as a signal for Machine Api Operator to deploy only provider-agnostic
```

Comment:
Would this also include a "capability" being enabled, as described in the "Conjunction with Capabilities" section above? Some providers may not want any machine API usage at all. Others may want to use CAPI instead, which we aren't very far away from being able to support.

Reply:
yes, this is good point. these component scenarios assume that the capability has been enabled for the component. should we make it more explicit? something like:

```suggestion
In the future, "External" platform type, in conjunction with an enabled capability, would serve as a signal for Machine Api Operator to deploy only provider-agnostic
```